tests/python/test_dataset_aggregation.py

```diff
@@ -0,0 +1,157 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2022, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES LOSS OF USE, DATA, OR PROFITS OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import numpy as np
+import pytest
+from smartredis import Client, Dataset
+from smartredis.error import *
+
+
+def test_aggregation(use_cluster):
+    client = Client(None, use_cluster)
+
+    # Build datasets
+    dataset_1 = create_dataset("dataset_1")
+    dataset_2 = create_dataset("dataset_2")
+    dataset_3 = create_dataset("dataset_3")
+    dataset_4 = create_dataset("dataset_4")
+
+    # Make sure the list is cleared
+    list_name = "dataset_test_list"
+    client.delete_list(list_name)
+
+    # Put two datasets into the list
+    client.put_dataset(dataset_1)
+    client.put_dataset(dataset_2)
+    client.append_to_list(list_name, dataset_1)
+    client.append_to_list(list_name, dataset_2)
+
+    # Put the final two datasets into the list
+    client.put_dataset(dataset_3)
+    client.put_dataset(dataset_4)
+    client.append_to_list(list_name, dataset_3)
+    client.append_to_list(list_name, dataset_4)
+
+    actual_length = 4
```

Comment:
Move this magic number to a default argument to the signature so that we can scale this test if we ever run into the problem for large numbers of datasets

Reply:
Can't meaningfully put it in the signature since this is a pytest test case, but I moved it to the top of the function as a single constant that's used throughout the test case so that updates can be made in one place